cpp/dolfinx/refinement/refine.h

```diff
@@ -72,7 +72,7 @@ refine(const mesh::Mesh<T>& mesh,
 
   mesh::Mesh<T> mesh1 = mesh::create_mesh(
       mesh.comm(), mesh.comm(), cell_adj.array(), mesh.geometry().cmap(),
-      mesh.comm(), new_vertex_coords, xshape, partitioner);
+      mesh.comm(), new_vertex_coords, xshape, mesh.geometry().dim(), partitioner);
```

Comment:
Shouldn’t xshape[1] and mesh.geometry().dim() be the same?

Reply:
Yes - but in terms of the code being correct in meaning, the former is the shape of the geometry data and the latter is the geometric dimension.

Reply:
Do we check if they match? We should probably add some sanity checks.